Services/Sources/GitHubSource.cs

```diff
@@ -20,11 +20,47 @@
 
 namespace Services.Sources
 {
+
+    public interface IGitHubSource : ISource
+    {
+
+        /// <summary>
+        /// Gets the source.
+        /// </summary>
+        /// <param name="uri">The URI of the source project.</param>
+        /// <exception cref="NotImplementedException"></exception>
+        new void GetSource(Uri uri);
```

Comment:
You don't need to implement the methods with new in the interface since they are already inherited from the parent interface. I didn't notice this in my first review, my apologies.
You can remove all the methods from this interface with the new method (yes, you'll get an empty interface, this is fine).
Also remove these from the IGitlabSource